### PR TITLE
fix: sidebar groups are not opened when directly navigating to these pages (fix #2564)

### LINKED ISSUE #2564: Calculate sidebar expansion for groups with paths on page open
- [x] I confirm that this is an issue rather than a question.

Sidebar groups with paths do not expand the sidebar when directly navigating to these pages.

#### Steps to reproduce

Demo repo: https://github.com/echojoshchen/vuepress-subheading-demo
1. Navigate to /test/sections, the first sidebar group expands instead of the active one.
2. Navigate to /test/section-2, the first sidebar group expands, and no active header is shown.

#### What is expected?

The sidebar should be expanded to the same layout as when navigating by clicking.

#### What is actually happening?

The first sidebar group expands instead of the active one.

#### Other relevant information

### PR DESCRIPTION
**Pull request by echojoshchen**
**Summary**

**What kind of change does this PR introduce?** (check at least one)

- [x] Bugfix
- [ ] Feature
- [ ] Code style update
- [ ] Refactor
- [ ] Docs
- [ ] Build-related changes
- [ ] Other, please describe:

If changing the UI of default theme, please provide the **before/after** screenshot:

**Does this PR introduce a breaking change?** (check one)

- [ ] Yes
- [x] No

If yes, please describe the impact and migration path for existing applications:

**The PR fulfills these requirements:**

- [x] When resolving a specific issue, it's referenced in the PR's title (e.g. `fix #xxx[,#xxx]`, where "xxx" is the issue number)

You have tested in the following browsers: (Providing a detailed version will be better.)

- [x] Chrome
- [x] Firefox
- [x] Safari
- [x] Edge
- [ ] IE

If adding a **new feature**, the PR's description includes:

- [ ] A convincing reason for adding this feature
- [ ] Related documents have been updated
- [ ] Related tests have been updated

To avoid wasting your time, it's best to open a **feature request issue** first and wait for approval before working on it.

**Other information:**

### CODE REVIEW

**Comment by echojoshchen** [comment]
#2564

**Comment by echojoshchen** [comment]
@billyyyyy3320 Thanks for taking a look at this PR! Is there anything blocking this from being merged?

**Comment by echojoshchen** [comment]
@billyyyyy3320 Could this PR be merged soon? It would help a lot for nested pages

File: packages/@vuepress/theme-default/components/SidebarLinks.vue [hunk 1]
```diff
@@ -96,7 +96,7 @@ function descendantIsActive (route, item) {
       } else {
         return child.type === 'page' && isActive(route, child.path)
       }
-    })
+    }) || item.path && isActive(route, item.path)
   }
```
Comment (billyyyyy3320) [inline packages/@vuepress/theme-default/components/SidebarLinks.vue:100]: Sorry for the very late review. Thanks for the PR, and the functionality looks good to me. 

Only a small suggestion, there're too many conditions without comments and variable names for others to understand. Would you like to try the code below

```js
function descendantIsActive (route, item) {
  if (item.type === 'group') {
    const childIsActive = item.path && isActive(route, item.path)
    const grandChildIsActive = item.children.some(child => {
      if (child.type === 'group') {
        return descendantIsActive(route, child)
      } else {
        return child.type === 'page' && isActive(route, child.path)
      }
    })

    return childIsActive || grandChildIsActive
  }
  return false
}
```

Comment (echojoshchen) [reply packages/@vuepress/theme-default/components/SidebarLinks.vue:103]: Thanks, I think that's much clearer. I've implemented your suggestion.

**Review by billyyyyy3320 — APPROVED** [review]
Thanks!

**Pull request merged**